101-lab/lab-notes.md

```diff
@@ -0,0 +1,130 @@
+
+- General Lab Intro Information
+    - lab is self-directed over about 2 weeks
+    - 3 meetings
+    - if you aren't finished in the next 2 weeks, let us know and we'll do what we can to keep your access to the namespaces a little longer
+    - point out and show off links to the lab content and content repo
+- RocketChat and GitHub Access
+    - Everyone sign into RocketChat and join the channel (show them how?)
+        - Mention the "new user sign-ups disabled" message, and to ignore it and click login anyway.
+    - Post your GitHub username in RocketChat
+    - Is everyone a member of the BCGov orgs in GitHub? If not, use the Just Ask link.
+- OpenShift Tour (while we set up everyone's RocketChat and GitHub Access)
+    - Tour of the OpenShift console
+    - How to download the CLI
+    - How to log in to the CLI
+    - Show the lab namespaces
+        - Talk about the namespace name (how the lab will sometimes use `ocp101-tools` or `[-tools]` and what to use instead)
+        - Mention that the attendees don't have access yet, not to worry!
+- Lab Step 1: Access Control
+    - Talk about the "team" format of the lab (aka most of you are devs, but we need a few devops specialists to be your admins)
+    - Get volunteer admins (at least two)
+    - Walk through the process of adding people to the namespace
+        - Mention the need to include `@github` and the fact that names are lowercase only
+    - Get the admins to work through the rest of the list
+- Tips and Tricks
+    - show off `oc explain` and `-h` flag
+    - mention the deployment vs deploymentconfig issue (there is a step where you're asked to issue an `oc rollout` command which won't work - you can skip this step)
+- Questions?
+
+## Mid-Lab Check-In
+
+- Questions? 
+- Talk about Infrastructure As Code
+    - TODO: Maybe let's build an example set of yaml files that produces everything from the lab? Then the teams can look at this and use it as an example!
+    - Why is Infrastructure as Code a good idea?
+    - How are yaml manifests laid out?
+    - How do I generate yaml files? How do I apply them?
+    - What about templates?
+
+## Wrap-Up
+
+- How far did everyone get?
+    - Should we leave the lab stuff up for you guys?
+- Questions?
+- Services and Best Practices (don't worry, there are links and notes that will be posted about this!)
+    - Databases
+        - Already used MongoDB - the second most popular DB on the cluster
+        - Postgres is the most popular, and can be implemented in a few ways:
+            - Patroni is most common; you can use our images in either `bcgov` or Artifactory!
+            - EDB Operator is for teams with a license - this nets you additional enterprise-level support. Talk to Olena!
+            - CrunchyDB offers both enterprise and community licenses. Still new to the cluster!
+        - Our community has built an awesome backup-container, super easy DB backups! Take advantage of it!
```

Comment:
I found that s2i-caddy is also a very useful tool and two people actually asked me if we have something like this. Do we want to mention it as well?